crates/resolver/src/name_server/name_server_pool.rs

```diff
@@ -167,7 +167,8 @@ where
             // select the highest priority connection
             //   reorder the connections based on current view...
             //   this reorders the inner set
-            ServerOrderingStrategy::QueryStatistics => conns.sort_unstable(),
+            ServerOrderingStrategy::QueryStatistics => conns
+                .sort_unstable_by(|a, b| a.stats.decayed_srtt().total_cmp(&b.stats.decayed_srtt())),
```

Comment:
For some reason I'm nervous about this, but not sure why. I guess there's no other location where we do this comparison, so this is probably fine.

Reply:
It's a little hard to argue with if you can't articulate why!

I feel fairly strongly that this is a good change. IMO `PartialOrd`/`Ord` implementations should have a fairly obvious meaning and I don't think such a meaning exists for the concept of a `NameServer`. Instead, callers should make explicit how/why they want to compare name servers, which this is doing.

Note that the previous version actually had a special case deferring to `PartialEq` (`if self == other { return Ordering::Equal }`) and `PartialEq` actually only compares `self.config` to `other.config` -- and the `NameServerConfig` `PartialEq` implementations actually only compared the socket address and protocol. All of that was left implicit and could be considered surprising -- for example, if you configured two name server configs using HTTP using different `http_endpoint`s those would be considered equal and we would not take observed latencies into account when comparing those servers.